x-pack/plugins/ml/public/application/model_management/test_models/test_flyout.tsx

```diff
@@ -41,90 +41,96 @@ export const TestTrainedModelFlyout: FC<Props> = ({ model, onClose }) => {
+                id="xpack.ml.trainedModels.testModelsFlyout.deploymentIdLabel"
+                defaultMessage="Deployment ID"
               />
-            </h2>
-          </EuiTitle>
-          <EuiSpacer size="s" />
-          <EuiTitle size="xs">
-            <h4>{model.model_id}</h4>
-          </EuiTitle>
-        </EuiFlyoutHeader>
-        <EuiFlyoutBody>
-          {model.deployment_ids.length > 1 ? (
-            <>
-              <EuiFormRow
-                fullWidth
-                label={
-                  <FormattedMessage
-                    id="xpack.ml.trainedModels.testModelsFlyout.deploymentIdLabel"
-                    defaultMessage="Deployment ID"
-                  />
-                }
-              >
-                <EuiSelect
-                  fullWidth
-                  options={model.deployment_ids.map((v) => {
-                    return { text: v, value: v };
-                  })}
-                  value={deploymentId}
-                  onChange={(e) => {
-                    setDeploymentId(e.target.value);
-                  }}
-                />
-              </EuiFormRow>
-              <EuiSpacer size="l" />
-            </>
-          ) : null}
-
-          {onlyShowTab === undefined ? (
-            <>
-              <EuiTabs
-                size="m"
-                css={{
-                  marginTop: `-${mediumPadding}`,
-                }}
-              >
-                <EuiTab
-                  isSelected={inputType === INPUT_TYPE.TEXT}
-                  onClick={() => setInputType(INPUT_TYPE.TEXT)}
-                >
-                  <FormattedMessage
-                    id="xpack.ml.trainedModels.testModelsFlyout.textTab"
-                    defaultMessage="Test using text"
-                  />
-                </EuiTab>
-                <EuiTab
-                  isSelected={inputType === INPUT_TYPE.INDEX}
-                  onClick={() => setInputType(INPUT_TYPE.INDEX)}
-                >
-                  <FormattedMessage
-                    id="xpack.ml.trainedModels.testModelsFlyout.indexTab"
-                    defaultMessage="Test using existing index"
-                  />
-                </EuiTab>
-              </EuiTabs>
-
-              <EuiSpacer size="m" />
-            </>
-          ) : null}
+            }
+          >
+            <EuiSelect
+              fullWidth
+              options={model.deployment_ids.map((v) => {
+                return { text: v, value: v };
+              })}
+              value={deploymentId}
+              onChange={(e) => {
+                setDeploymentId(e.target.value);
+              }}
+            />
+          </EuiFormRow>
+          <EuiSpacer size="l" />
+        </>
+      ) : null}
+      {onlyShowTab === undefined ? (
+        <>
+          <EuiTabs
+            size="m"
+            css={{
+              marginTop: `-${mediumPadding}`,
+            }}
+          >
+            <EuiTab
```

Comment:
In the 'Test' step in the wizard, does 'Test using text' correctly use the inference config if edited in the Details step? I tried editing `num_top_classes` and always get the default of 5, whereas 'Test using index' correctly returns the number of classes from the edited config.

```
"inference_config": {
          "classification": {
            "num_top_classes": 3
          }
        }
```

Reply:
Good catch - updated the text infer method to use the creation pipeline config in 8e0c0ee7f9fc9d90db5838cb22de9c97fe1e7c46